common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,121 @@
+"""OLI analytics service event tracker backend."""
+
+from __future__ import absolute_import
+
+import json
+import logging
+
+from requests_oauthlib import OAuth1Session
+
+from django.contrib.auth.models import User
+
+from student.models import anonymous_id_for_user
+from track.backends import BaseBackend
+
+
+LOG = logging.getLogger(__name__)
+
+
+class OLIAnalyticsBackend(BaseBackend):
+    """
+    Transmit events to the OLI analytics service
+    """
+    def __init__(self, **kwargs):
+        super(OLIAnalyticsBackend, self).__init__(**kwargs)
+
+        # the oli analytics service endpoint will be at
+        # "{}{}".format(oli_analytics_service_url, path)
+        self.url = kwargs.get('url', None)
```

Comment:
`None` is actually the default here; we can omit it.

`self.url = kwargs.get('url')`

Alternatively, you could just make the kwargs named parameters like:
```python
def __init__(
    self,
    url=None,
    path=None,
    key=None,
    secret=None,
    course_ids=None,
    **kwargs
):
    """
    Note: for mutable data types, like set/dict/list, you cannot do:
    course_ids=[]
    as that reference will be shared by all invocations of the method.

    If a mutable object is passed, the pattern is to default to None and
    then handle the special case as below:
    """
    self.url = url
    self.path = path
    self.key = key
    self.secret = secret
    self.course_ids = set()
    if course_ids is not None:
        self.course_ids = set(course_ids)
```

A benefit to using named parameters is that improves documentation; many tools and IDEs will autogenerate documentation based on method signatures. They are unable to do this when grabbing directly from `kwargs`.